src/HexagonOptimize.cpp

```diff
@@ -162,6 +162,8 @@ std::vector<Pattern> casts = {
     // Saturating narrowing casts
     { "halide.hexagon.satub.shr.vh.h", u8c(wild_i16x/wild_i16), Pattern::DeinterleaveOps | Pattern::ExactLog2Op1 },
     { "halide.hexagon.satuh.shr.vw.w", u16c(wild_i32x/wild_i32), Pattern::DeinterleaveOps | Pattern::ExactLog2Op1 },
+    { "halide.hexagon.sath.shr.vw.w", i16c(wild_i32x/wild_i32), Pattern::DeinterleaveOps | Pattern::ExactLog2Op1 },
+    { "halide.hexagon.satub.shr.vw.w", u8c(wild_i32x/wild_i32), Pattern::DeinterleaveOps | Pattern::ExactLog2Op1 },
```

Comment:
Actually, my comment about the two deinterleavings, one being vw, one being vh is missing some details. Here's what I think the IR needs to look like for a double narrowing:

    y = u8(deinterleave.vh(u16(deinterleave.vw(x_i32))))

As opposed to:

    y = u8(u16(deinterleave.vw(x_i32))

Which is I *think* what you are trying to do right now.